mets_mods2tei/api/tei.py

```diff
@@ -330,26 +336,70 @@ def bibl(self):
 
     def set_main_title(self, string):
         """
-        Set the main title of the title statements.
+        Set the main title of the tei:titleStmt.
         """
-        for main_title in self.tree.xpath('//tei:titleStmt/tei:title[@type="main"]', namespaces=ns):
-            main_title.text = string
+        titleStmt = self.tree.xpath('//tei:titleStmt', namespaces=ns)[0]
+        for node in titleStmt.xpath('tei:title[@type="main"]', namespaces=ns):
+            node.text = string
 
-    def set_publication_level(self, level):
+    def add_sub_title(self, string):
```

Comment:
The [single commit](https://github.com/slub/mets-mods2tei/pull/58/commits/93fb68469356779ba73d08c05dc6572542e61372) is readable IMO.

Here, `set_publication_level` is obsolete (because it is already mapped from the `div/@TYPE` during `__spur()`, and `add_sub_title` as well as `add_part_title` is new.